core/sparse_array.cpp

```diff
@@ -95,6 +141,31 @@ void SparseArray::Reset(unsigned int p_nrows, unsigned int p_ncols)
 	finished_ = false;
 }
 
+void SparseArray::Resize(uint32_t p_row)
+{
+    //vectors are reallocated by 1000 entries at a time. It can be wasteful but probably better than making a function call for each new entry. Will defer the final decision to Ben. 
+
+
+    *(columns + p_row)=(uint32_t *)realloc(*(columns + p_row), 1000 * sizeof(uint32_t));  
```

Comment:
This should be adding to the existing size, not just reallocing at a fixed size of 1000*sizeof(uint32_t), right?  Probably the only reason this code doesn't crash is that the model you're testing with probably never causes a realloc.  :->